Content.Shared/Storage/ItemCounter/SharedItemCounterSystem.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections.Generic;
+using JetBrains.Annotations;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+
+namespace Content.Shared.Storage.ItemCounter
+{
+    [UsedImplicitly]
+    public abstract class SharedItemCounterSystem : EntitySystem
+    {
+        /// <inheritdoc />
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeNetworkEvent<AfterStorageFillEvent>(OnStorageFill);
+            SubscribeLocalEvent<SharedItemCounterComponent, EntInsertedIntoContainerMessage>(HandleEntityInsert);
+            SubscribeLocalEvent<SharedItemCounterComponent, EntRemovedFromContainerMessage>(HandleEntityRemoved);
+        }
+
+        private void OnStorageFill(AfterStorageFillEvent args)
+        {
+            if (ComponentManager.TryGetComponent(args.Entity.Uid, out SharedAppearanceComponent appearance))
+            {
+                var allLayer = new List<EntityUid>();
+                foreach (var entity in args.ContainedEntities)
+                {
+                    allLayer.Add(entity.Uid);
+                }
+
+                appearance.SetData(StorageMapVisuals.AllLayers, new ListOfUids(allLayer));
+            }
+        }
```

Comment:
Why wouldn't the server just subscribe to this locally? Appearance data would get synced to the client anyway?